server/src/features/auth/controller.ts

```diff
@@ -1,36 +1,79 @@
-import { Body, Controller, Post, Route, Tags } from "tsoa";
+import { Body, Controller, Middlewares, Post, Request, Response, Route, SuccessResponse, Tags } from "tsoa";
 import { User } from "@prisma/client";
 import { AuthService } from "./services";
 import { UserCreationParams, UserLoginParams } from "../users/model";
+import cookieParser from "cookie-parser";
+import express, {
+  Response as ExResponse,
+  Request as ExRequest,
+  NextFunction,
+  RequestHandler,
+} from "express";
+import jwt from "jsonwebtoken";
+import { SECRET_KEY } from "../../../config";
+import { AuthError } from "../../helpers/errors";
+import { createAccessToken, decodeAuthToken } from "../../helpers/jwt";
+import { UserService } from "../users/services";
 
-@Route("login")
+export interface UserInfo {
+  id: number,
+  accessToken: string,
+  refreshToken: string
+}
+//@Middlewares([cookieParser])
+@Route("session")
 @Tags("Auth")
-export class LoginController extends Controller {
+export class AuthController extends Controller {
   /**
    * Logs in user to the system and creates an access token.
    * @param requestBody Login (email) and password
    * @returns User authentication details
    */
-  @Post()
+  @Post("authenticate")
   public async login(
+    @Request() req: ExRequest,
     @Body() requestBody: UserLoginParams
-  ): Promise<User | string> {
-    return new AuthService().login(requestBody);
+    ): Promise<Partial<User>> {
+      const data = await new AuthService().getUserId(requestBody);
+      const refreshToken = await new AuthService().generateRefreshToken(data.id, data.role);
+      req.res?.cookie('refresh_token', refreshToken, { 
+        httpOnly: true,
+        path: '/session/refresh',
+      })
```

Comment:
refreshToken business should probably be extracted in a separate function called here and in route bellow (start)